cli/js/remove_test.ts

```diff
@@ -169,7 +169,13 @@ testPerm({ write: true, read: true }, function removeAllSyncDirSuccess(): void {
   assert(pathInfo.isDirectory()); // check exist first
   const subPathInfo = Deno.statSync(subPath);
   assert(subPathInfo.isDirectory()); // check exist first
-  Deno.removeSync(path, { recursive: true }); // remove
+  let success = true;
+  try {
+    Deno.removeSync(path, { recursive: true }); // remove
+  } catch (e) {
+    success = false;
+  }
+  assert(success);
```

Comment:
What's the purpose of this try-catch if it's supposed to succeed?

Reply:
Yeah, this PR has gotten confusing as it's evolved. It might be better for me to just close it and re-push a new version of it with some of these commits squashed together. But for explanation, here's what's going on/how we got here. Other than the `debug!` improvements, there's not much else in this PR except a bit of refactoring. One thing I did with the refactoring was move the third-party imports to `cli/fs.rs`, because I thought that was the guiding intent about how to divide work between `cli/fs.rs` and `cli/ops/fs.rs`. You explained in an earlier comment thread on this PR that no, things are probably just divided as they are because of the contingencies of past refactorings, and stuff should ideally all be in `cli/ops/fs.rs` if it can be. So I pushed revisions to this branch that moved things back. Also, @bartlomieju noticed that one of the imports I was moving back and forth between the files was of the third-party crate `remove_dir_all`. He said, we don't need that, there's such a function in `std::fs`. I hadn't noticed that, so I tried replacing the use of `remove_dir_all::remove_dir_all` with a plain `std::fs::remove_dir_all`. That turned out to work fine on Mac and Unix, but when I pushed it to the branch this PR is based on, the CI test for Windows turned out to fail. (See above comment exchange between me and @bartlomieju.) So I pushed another revision that re-instated the use of `remove_dir_all::remove_dir_all`, at least for the time being until we worked out a good solution for all platforms. Also, I noticed that the report of the failure on Windows was funny: it didn't say that a specific test failed, but there was just a `error: Uncaught Error: The directory is not empty. (os error 145).` I don't understand why. I tried adding the `try ... catch` blocks to those tests, so that if someone else breaks this in the same way later, they'd get a better error report. But maybe those fixes are stupid, and would fail in just the same way that the bare `remove` call did.

Anyway, hope that explains what's going on. Should I close this PR and resubmit a squashed/pruned version of it?

Reply:
@ry, check out [this branch on my fork](https://github.com/dubiousjim/deno/commits/filesystem1bb). That has six commits on top of denoland/master, the first four being the good parts of this PR, and the last two replacing the use of `remove_dir_all::remove_dir_all` with `std::fs::remove_dir_all`. Based on what happened before, I'm expecting the branch to fail CI tests on Windows, and so the last two commits should be reverted/dropped. That would be the pruned version of this PR.

Reply:
Confirmed that that fails, with only this information:

> error: Uncaught Error: The directory is not empty. (os error 145)
> $deno$/dispatch_json.ts:41:11
>     at unwrapResponse ($deno$/dispatch_json.ts:41:11)
>     at sendAsync ($deno$/dispatch_json.ts:96:10)
> test tests::std_tests ... FAILED
> 
> failures:
> 
> ---- tests::std_tests stdout ----
> target_dir D:\a\deno\deno\target\release
> ##[error]thread 'tests::std_tests' panicked at 'assertion failed: status.success()', cli\tests\std_tests.rs:31:5
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
> error: test failed, to rerun pass '-p deno --test std_tests'
> 
> failures:
>     tests::std_tests
> 
> test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
> 
> ##[error]Process completed with exit code 1.

Reply:
Fails only on windows.